src/include/mpir_mem.h

```diff
@@ -136,67 +135,51 @@ extern "C" {
 
 
 /* Persistent memory that we may want to recover if something goes wrong */
-#define MPIR_CHKPMEM_DECL(n_)                                   \
-    void *(mpiu_chkpmem_stk_[n_]) = { NULL };                   \
-    int mpiu_chkpmem_stk_sp_=0;                                 \
-    MPIR_AssertDeclValue(const int mpiu_chkpmem_stk_sz_,n_)
-#define MPIR_CHKPMEM_MALLOC_ORSTMT(pointer_,type_,nbytes_,rc_,name_,class_,stmt_) \
-    {                                                                   \
-        pointer_ = (type_)MPL_malloc(nbytes_,class_);                   \
-        if (pointer_) {                                                 \
-            MPIR_Assert(mpiu_chkpmem_stk_sp_<mpiu_chkpmem_stk_sz_);     \
-            mpiu_chkpmem_stk_[mpiu_chkpmem_stk_sp_++] = pointer_;       \
-        } else if (nbytes_ > 0) {                                       \
-            MPIR_CHKMEM_SETERR(rc_,nbytes_,name_);                      \
-            stmt_;                                                      \
-        }                                                               \
-    }
-#define MPIR_CHKPMEM_REGISTER(pointer_)                         \
-    {                                                           \
-        MPIR_Assert(mpiu_chkpmem_stk_sp_<mpiu_chkpmem_stk_sz_); \
-        mpiu_chkpmem_stk_[mpiu_chkpmem_stk_sp_++] = pointer_;   \
-    }
-#define MPIR_CHKPMEM_REAP()                                             \
-    {                                                                   \
-        while (mpiu_chkpmem_stk_sp_ > 0) {                              \
-            MPL_free(mpiu_chkpmem_stk_[--mpiu_chkpmem_stk_sp_]);        \
-        }                                                               \
-    }
-#define MPIR_CHKPMEM_COMMIT()                   \
+#define MPIR_CHKPMEM_MAX 10
+#define MPIR_CHKPMEM_DECL() \
+    void *(mpiu_chkpmem_stk_[MPIR_CHKPMEM_MAX]) = { NULL }; \
+    int mpiu_chkpmem_stk_sp_=0;
+
+#define MPIR_CHKPMEM_MALLOC(pointer_,nbytes_,class_) \
+    do { \
+        pointer_ = MPL_malloc(nbytes_,class_); \
+        if (pointer_) { \
+            MPIR_Assert(mpiu_chkpmem_stk_sp_<MPIR_CHKPMEM_MAX); \
+            mpiu_chkpmem_stk_[mpiu_chkpmem_stk_sp_++] = pointer_; \
+        } else if (nbytes_ > 0) { \
+            MPIR_ERR_SETANDJUMP(mpi_errno, MPI_ERR_OTHER, "**nomem"); \
+        } \
+    } while (0)
+
+#define MPIR_CHKPMEM_ADD(pointer_) \
```

Comment:
What difference is trying to be expressed by this rename?

Reply:
Not much. I find `_ADD` more intuitive (and easier to type) than `_REGISTER`.  If we go back, then I need rename `MPIR_CHKLMEM_ADD` as well.

Reply:
`CHKLMEM_ADD` is not used unless I'm missing something.
```console
raffenet@csi0363877:main/ (2412_CHKLMEM*) $ git grep CHKLMEM_ADD
src/include/mpir_mem.h:#define MPIR_CHKLMEM_ADD(pointer_) \
```

Reply:
I used `MPIR_CHKLMEM_ADD` in my other Work-in-progress :) Anyway, if `_REGISTER` is preferred, I can change.

Reply:
Let's leave it as REGISTER then.

Reply:
Changed both names to use `MPIR_CHK[LP]MEM_REGISTER`.

Added a CHANGES entry.